src/main/java/uk/gov/companieshouse/api/interceptor/MappedCRUDAuthenticationInterceptor.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.companieshouse.api.interceptor;
+
+import org.springframework.http.HttpMethod;
+import uk.gov.companieshouse.api.util.security.Permission;
+import uk.gov.companieshouse.api.util.security.TokenPermissions;
+
+/**
+ * Replacement for {@link CRUDAuthenticationInterceptor} implemented with
+ * {@link MappablePermissionsInterceptor}.
+ * Checks the request contains the relevant token permission values based on the HTTP method.
+ * It will try to find a {@link TokenPermissions} object in the request or create one and store
+ * it in the request if not.
+ */
+public class MappedCRUDAuthenticationInterceptor extends MappablePermissionsInterceptor {
+    private static final PermissionsMapping CRUD_MAPPING = PermissionsMapping.builder()
+            .defaultAllOf(Permission.Value.READ)
+            .mapAllOf(HttpMethod.PUT.toString(), Permission.Value.UPDATE)
+            .mapAllOf(HttpMethod.PATCH.toString(), Permission.Value.UPDATE)
+            .mapAllOf(HttpMethod.POST.toString(), Permission.Value.CREATE)
+            .build();
```

Comment:
Not sure if I am reading this correct, so this might not be relevant, but if this one is replicating the existing CRUDAuthenticationInterceptor, would it be worth adding the DELETE into the builder too?

Reply:
Good spot, my mistake -- i'll add DELETE to the mapping.